### ISSUE TITLE
Different local %%opts in Layout

### ISSUE THREAD

**Issue by basnijholt** [ioam/holoviews#93]
I would like to have the ability to set different styling options for different Images in a ``Layout``.

I would expect things to work like this:
```python
%%opts Image (cmap='gist_heat_r')
hv.Image(np.random.rand(10,10))(plot=dict(xaxis=None, yaxis=None), style=dict(cmap='jet')) + hv.Image(np.random.rand(10,10))
```
 and produce:
![download](https://cloud.githubusercontent.com/assets/6897215/7769490/6f8e08f2-0084-11e5-8e26-66f887d504ed.png)

but it actually produces:

![download 1](https://cloud.githubusercontent.com/assets/6897215/7769486/67f932d8-0084-11e5-927c-9c225eddd58b.png)

**Comment by vascotenner** [comment]
I would solve this by using groups:

```python
%%opts Image.noaxis [xaxis=None, yaxis=None, show_title=False] (cmap='jet') Image (cmap='gist_heat_r') 
hv.Image(np.random.rand(10,10), group='noaxis') + hv.Image(np.random.rand(10,10))
```

**Comment by jlstevens** [comment]
I agree that the groups solution is the safest approach! :-)

Note that if you want Images without axes often, you can just use the ``%opts`` line magic, allowing you to define it once so you can reuse it whenever you like by simply assigning the group:

```python
%opts Image.noaxis [xaxis=None, yaxis=None, show_title=False]
```
I will now explain what is and isn't a bug in your example (because there is a bug!):

* The fact that the two color maps are the same is not a bug: the ``%%opts`` magic always takes precedence (something we should document better!) which means all Images are using the 'gist_heat_r' map as requested (``cmap='jet'`` is simply overridden). As Vasco points out, you can get around this easily by using groups.

* The fact that ``xaxis=None`` and ``yaxis=None`` isn't being respected *is* a bug. As these options aren't being specified in the ``%%opts`` magic, they aren't being overwritten which means they *should* be applied.

Hopefully that helps explains what is going on here!

 I hope to have a look at the bug in the options system tonight.

**Comment by jlstevens** [comment]
These bug reports are really forcing us to get things right! :-)

We think these bugs ares all fixed in commits cb40361 through to 9d3cfee and we have a new unit test to make sure the bad behavior doesn't reappear. In short, the order in which you specified options was causing different sets of option keywords to be returned to the plotting system.

These commits also fix #96 so I shall now closes this issue. If after pulling the latest version the problem still isn't fixed, please report it and I will reopen this issue!

**Comment by jlstevens** [comment]
Note that we needed to make a few more fixes after the commits mentioned. As of ff37b55 all the tests are passing and we think it is ready to use!
